torch/_inductor/mkldnn_lowerings.py

```diff
@@ -529,6 +529,41 @@ def qlinear_unary(
             else:
                 x_zp.realize()
 
+            def _is_act_zp_compatible_with_onednn():
+                # oneDNN doesn't currently have an optimized kernel for a vector activation zero-point.
+                # So, if the activation zero point is a vector, we won't support it with oneDNN.
+                # We can support the provided activation if:
+                # 1. if x_zp is dummy (empty tensor) created to match an Inductor pattern
+                # 2. If the activation is per-tensor quantized (x may even be uint8 in that case).
+                # in the latter case, the quantization may either be static or dynamic.
+                nonlocal x_zp
+                if (
+                    isinstance(
+                        ir.InputsKernel.unwrap_storage_for_input(x_zp),
+                        ir.ComputedBuffer,
+                    )
+                    and (
+                        (x_zp.get_numel() == 0 and x.get_dtype() == torch.int8)
+                        or (  # dummy x_zp
+                            x_zp.get_numel() == 1
+                        )  # per-tensor dynamic quantization of x
+                    )
+                ) or (
+                    isinstance(
+                        ir.InputsKernel.unwrap_storage_for_input(x_zp),
+                        ir.ConstantBuffer,
+                    )
+                    and len(x_zp.get_layout().size)
+                    == 0  # statically quantized per-tensor quantization of x
```

Comment:
nit: `static per-tensor quantization of x (asymmetric/symmetric in case of uint8, asymmetric in case of int8)`